src/main/org/firebirdsql/jca/FBResourceException.java

```diff
@@ -81,7 +81,7 @@ public FBResourceException(String reason, Exception original) {
         super(reason, SQLStateConstants.SQL_STATE_GENERAL_ERROR);
         // Preserve setLinkedException for backwards compatibility
         setLinkedException(original);
-        initCause(original);
+        if (getCause() != original) initCause(original);
```

Comment:
This should be `if (getCause() == null) ...`

Reply:
Throwable constructor initalizes cause to this, not null. And suprisingly initCause throws an exception even if you set the same value twice.
But you are right, this still can trigger an exception.
Better condition is ` if (getCause() == this) ... `

Reply:
Although the `cause` field is initialized with this, The method `getCause()` actually returns `null` in that situation:

```
public synchronized Throwable getCause() {
    return (cause==this ? null : cause);
}
```